### DISCUSSION TITLE
How to mock a HostAttributeToken?

### DISCUSSION THREAD

**Discussion by jase88** [help-me-mom/ng-mocks#9663] [Q&A] [answered]
Hi,

Angular allows to set static attributes to components/directives.
For example:
```typescript
@Directive()
class MyDir {
  attr: string = inject(new HostAttributeToken('some-attr'));
}
```
see https://angular.dev/api/core/HostAttributeToken?tab=usage-notes

__How would I mock a _HostAttributeToken___ with _ng-mocks_?

For non optional host attribute tokens the test would error with:

> Error: NG0201: No provider for HostAttributeToken some-attr found. Find more at https://angular.io/errors/NG0201

**Comment by satanTime** [comment, marked as answer]
A `HostAttributeToken` should not be mocked as a provider. Angular resolves it from a static attribute on the directive or component host node, so render an explicit host template containing that attribute.

For a directive whose selector is `[myDir]`:

```ts
beforeEach(() => MockBuilder(MyDir));

it('injects the host attribute', () => {
  MockRender('<div myDir some-attr="expected"></div>');

  const directive = ngMocks.findInstance(MyDir);
  expect(directive.attr).toBe('expected');
});
```

For a component, use the same pattern with its element selector, for example `<my-component some-attr="expected"></my-component>`. This supplies the static host attribute through Angular's normal injection mechanism.

Closing as answered.
